diff --git a/ldso/dynlink.c b/ldso/dynlink.c
index f23cf4a0..97f25696 100644
--- a/ldso/dynlink.c
+++ b/ldso/dynlink.c
@@ -1616,7 +1616,7 @@ static void do_init_fini(struct dso **queue)
 	pthread_mutex_unlock(&init_fini_lock);
 }
 
-void __libc_start_init(void)
+weak void __libc_start_init(void)
 {
 	do_init_fini(main_ctor_queue);
 	if (!__malloc_replaced && main_ctor_queue != builtin_ctor_queue)
@@ -1630,7 +1630,7 @@ static void dl_debug_state(void)
 
 weak_alias(dl_debug_state, _dl_debug_state);
 
-void __init_tls(size_t *auxv)
+weak void __init_tls(size_t *auxv)
 {
 }
 
